rackcity/views/asset_views.py

```diff
@@ -78,6 +79,26 @@ def asset_many(request):
     assets are returned. If page is specified as a query parameter, page
     size must also be specified, and a page of assets will be returned.
     """
+    if 'change_plan' in request.data:
+        change_plan_id = request.data['change_plan']
+        try:
+            change_plan = ChangePlan.objects.get(id=change_plan_id)
+        except ObjectDoesNotExist:
+            return JsonResponse(
+                {
+                    "failure_message":
+                        Status.ERROR.value +
+                        "Change Plan" + GenericFailure.DOES_NOT_EXIST.value,
+                    "errors":
+                        "No existing change plan with id="+str(change_plan_id)
+                },
+                status=HTTPStatus.BAD_REQUEST
+            )
```

Comment:
I have a helper function that does basically this ill do the refactor on my branch!

Reply:
we <3 refactoring

Reply:
we </3 rewriting